src/simcore_service_deployment_agent/portainer.py

```diff
@@ -1,48 +1,86 @@
+import asyncio
 import json
 import logging
-from typing import Dict, List
+import time
+from typing import Dict, List, Optional
 
-from aiohttp import ClientSession
+from aiohttp import ClientSession, ClientTimeout
+from tenacity import retry, stop_after_attempt, wait_fixed, wait_random
 from yarl import URL
 
-from .exceptions import ConfigurationError, AutoDeployAgentException
+from .exceptions import AutoDeployAgentException, ConfigurationError
 
 log = logging.getLogger(__name__)
 
-async def _portainer_request(url: URL, app_session: ClientSession, method: str, **kwargs) -> str:
-    async with getattr(app_session, method.lower())(url, **kwargs) as resp:
+NUMBER_OF_ATTEMPS = 5
+MAX_TIME_TO_WAIT_S = 10
+
+
+@retry(
+    stop=stop_after_attempt(NUMBER_OF_ATTEMPS),
+    wait=wait_fixed(3) + wait_random(0, MAX_TIME_TO_WAIT_S),
```

Comment:
would add reraise=True and before_sleep=before_sleep_log(log, logging.WARNING),

this way we get more info